pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -505,14 +505,15 @@ def _update_learning_rates(
                 self.scheduler_progress.increment_ready()
 
                 # update LR
-                if lr_scheduler["reduce_on_plateau"]:
-                    lr_scheduler["scheduler"].step(monitor_val)
-                else:
-                    lr_scheduler["scheduler"].step()
-
+                self.trainer._call_lightning_module_hook(
+                    "lr_scheduler_step",
+                    lr_scheduler["scheduler"],
+                    optimizer_idx=lr_scheduler["opt_idx"],
+                    monitor_val=monitor_val,
+                )
                 self.scheduler_progress.increment_completed()
 
-    def _get_monitor_value(self, key: str) -> Any:
+    def _get_monitor_value(self, key: str) -> Optional[Union[float, torch.Tensor]]:
```

Comment:
Technically nothing forbids that the user adds custom metrics of any type to the dictionary. Is the strict typing necessary?